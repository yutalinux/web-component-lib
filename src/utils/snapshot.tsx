import { type Attributes, type FunctionComponent, createElement } from "react";
import { renderToString } from "react-dom/server";

export function snapshotElement<T extends object>(
  type: FunctionComponent<T>,
  args: (Attributes & T) | null,
) {
  const element = createElement(type, args);
  return renderToString(element);
}
