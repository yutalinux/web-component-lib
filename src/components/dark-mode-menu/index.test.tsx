import { snapshotElement } from "@/utils/snapshot";
import { expect, test } from "vitest";
import { DarkModeButton } from ".";

test("Button Snapshot", () => {
  expect(snapshotElement(DarkModeButton, { label: "Button" })).toMatchSnapshot();
  expect(
    snapshotElement(DarkModeButton, { size: "small", label: "Button" }),
  ).toMatchSnapshot();
  expect(
    snapshotElement(DarkModeButton, { size: "large", label: "Button" }),
  ).toMatchSnapshot();
});
