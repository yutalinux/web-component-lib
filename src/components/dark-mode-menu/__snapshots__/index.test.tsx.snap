// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button Snapshot 1`] = `"<button type="button" aria-haspopup="true" aria-expanded="false" id="react-aria-:R0:" class="rounded-md outline-none border box-border border-neutral100 text-black bg-white hover:bg-neutral-100 dark:bg-neutral-900 dark:text-white dark:border-neutral-800 dark:hover:bg-neutral-800 text-base px-1 py-1" data-rac=""><svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 24 24" class="w-6 h-6" height="1em" width="1em" xmlns="http://www.w3.org/2000/svg"><path d="M20 3H4c-1.1 0-2 .9-2 2v11c0 1.1.9 2 2 2h3l-1 1v2h12v-2l-1-1h3c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 13H4V5h16v11z"></path></svg></button>"`;

exports[`Button Snapshot 2`] = `"<button type="button" aria-haspopup="true" aria-expanded="false" id="react-aria-:R0:" class="rounded-md outline-none border box-border border-neutral100 text-black bg-white hover:bg-neutral-100 dark:bg-neutral-900 dark:text-white dark:border-neutral-800 dark:hover:bg-neutral-800 text-sm px-1 py-1" data-rac=""><svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 24 24" class="w-6 h-6" height="1em" width="1em" xmlns="http://www.w3.org/2000/svg"><path d="M20 3H4c-1.1 0-2 .9-2 2v11c0 1.1.9 2 2 2h3l-1 1v2h12v-2l-1-1h3c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 13H4V5h16v11z"></path></svg></button>"`;

exports[`Button Snapshot 3`] = `"<button type="button" aria-haspopup="true" aria-expanded="false" id="react-aria-:R0:" class="rounded-md outline-none border box-border border-neutral100 text-black bg-white hover:bg-neutral-100 dark:bg-neutral-900 dark:text-white dark:border-neutral-800 dark:hover:bg-neutral-800 text-lg px-1 py-1" data-rac=""><svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 24 24" class="w-6 h-6" height="1em" width="1em" xmlns="http://www.w3.org/2000/svg"><path d="M20 3H4c-1.1 0-2 .9-2 2v11c0 1.1.9 2 2 2h3l-1 1v2h12v-2l-1-1h3c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 13H4V5h16v11z"></path></svg></button>"`;
